File: flutter/complex_app/lib/services/mock_user_service.dart
import '../models/user.dart';

class MockUserService {
  static final List<User> _users = List.generate(
    100,
    (i) => User(
      id: '${i + 1}',
      name: 'User${i + 1}',
      email: 'user${i + 1}@example.com',
    ),
  );

  static List<User> getUsers() => _users;
  static User? getUserById(String id) {
    try {
      return _users.firstWhere((u) => u.id == id);
    } catch (e) {
      return null;
    }
  }
}

File: kotlin_multiplatform/complex_app/composeApp/src/commonMain/kotlin/com/example/complex_app/services/MockUserService.kt
package com.example.complex_app.services

import com.example.complex_app.models.User

object MockUserService {
    private val users = List(100) {
        User(
            id = it.toString(),
            name = "User $it"
        )
    }
    fun getUser(id: String): User? = users.find { it.id == id }
    fun getAllUsers(): List<User> = users
}
